app/controllers/events_controller.rb

```diff
@@ -51,6 +59,21 @@ def set_event
       @event = Event.find(params[:id])
     end
 
+    # We receive start_date/end_date pairs in a weird format because forms
+    # are limited in the way they can arrange data for us. So we translate
+    # pairs of { start_date: [{day, month, year}, {day, month, year}], end_date: [...] }
+    # to the expected format [{start_date, end_date}, ...]
+    #
+    # @return array of start_date, end_date pairs
+    def date_range_params
+      dateRanges = params[:date_ranges] || {start_date: [], end_date: []}
+
+      dateRanges[:start_date].zip(dateRanges[:end_date]).map do |s, e|
+        DateRange.new(start_date: Date.new(s[:year].to_i, s[:month].to_i, s[:day].to_i),
```

Comment:
Date conversion ( `Date.new(...)` )might be refactored into a separate function to improve readability